VectorsCPP/Vector.hpp

```diff
@@ -0,0 +1,207 @@
+#pragma once
+#include <iostream>
+#include <initializer_list>
+#include <algorithm>
+
+constexpr int capacityNumber = 5;
+constexpr int multiplier = 2;
+
+template<typename T>
+class Vector
+{
+
+public:
+
+    // Constructors
+    /// <summary>
+    /// a constructor class
+    /// to handle vector()
+    /// </summary>
+    Vector()
+        : size(0),
+        capacity(capacityNumber),
+        array(new T[capacity])
+    {
+    }
+
+    /// <summary>
+    /// Constructor function to handle Vector(Size)
+    /// </summary>
+    /// <param name="rhs"></param>
+    Vector(const Vector& rhs)
+        : size(rhs.size),
+        capacity(rhs.capacity),
+        array(new T[capacity])
+    {
+        for (int i = 0; i < rhs.size; i++)
+        {
+            array[i] = rhs.array[i];
+        }
+    }
+
+    /// <summary>
+    /// Constructor function to handle vector(number, size)
+    /// </summary>
+    /// <param name="elements"></param>
+    /// <param name="value"></param>
+    Vector(int elements, T value)
+        : size(elements),
+        capacity(elements + capacityNumber),
+        array(new T[capacity])
+    {
+        for (int i = 0; i < size; i++)
+        {
+            array[i] = value;
+        }
+    }
+
+    /// <summary>
+    /// Constructor function to handle Vector {1,2,3,4,5}
+    /// </summary>
+    /// <param name="list"></param>
+    Vector(const std::initializer_list<T>& list)
+        : size(0),
+        capacity(list.size() + capacityNumber),
+        array(new T[capacity])
+    {
+        for (const T& i : list)
+        {
+            pushBack(i);
+        }
+    }
+
+    /// <summary>
+    /// Deconstructor for Vector()
+    /// </summary>
+    ~Vector()
+    {
+        delete[] array;
+    }
+
+    /// <summary>
+    /// Function to put a value at the top of the vector
+    /// </summary>
+    /// <param name="value"></param>
+    void pushBack(T value)
+    {
+        if (size < capacity)
+        {
+            array[size] = value;
+            ++size;
+        }
+        else
+        {
+            capacity *= multiplier;
+            T* newarray = new T[capacity];
+            for (int i = 0; i < size; i++)
+            {
+                newarray[i] = array[i];
```

Comment:
you're copying here a lot of data
might as well you `std::copy` or use `std::move` if you want to be more advanced